tests/TestCase.php

```diff
@@ -55,4 +60,48 @@ protected function psr7Response(array $content = null): ResponseInterface
 
         return $response;
     }
+
+    /**
+     * Return a HttpFoundation request with the provided content.
+     *
+     * @param  string $uri
+     * @param  string $method
+     * @param  array $content
+     * @return Request
+     */
+    protected function httpFoundationRequest(string $uri, string $method, array $content = null): Request
+    {
+        return Request::create(
+            self::BASE_URI . $uri,
+            $method,
+            [],
+            [],
+            [],
+            $content ? ['CONTENT_TYPE' => 'application/json'] : [],
+            $content ? json_encode($content) : ''
+        );
+    }
+
+    /**
+     * Return a PSR-7 server request with the provided content.
+     *
+     * @param  array $content
+     * @return ServerRequestInterface
+     */
+    protected function psr7ServerRequest(
+        string $uri,
+        string $method,
+        array $content = null
+    ): ServerRequestInterface {
+        $psr17Factory = new Psr17Factory();
+        $uri = $psr17Factory->createUri(self::BASE_URI . $uri);
+        $stream = $psr17Factory->createStream(json_encode($content));
+        $request = $psr17Factory->createServerRequest($method, $uri);
```

Comment:
Nitpicking again but yeah, I align equal signs everywhere so it's best to do it here too for consistency.

I will probably add a GitHub workflow to enforce coding style

Reply:
no worries, did not recognize the alignment in a first place ;-)